File: projects/cpp_hello_world_lib/greeter_test.cpp
#include "greeter.h"
#include <gtest/gtest.h>

TEST(GreeterTest, DefaultGreet) {
  EXPECT_EQ(greet(), "Hello World");
}

TEST(GreeterTest, CustomGreet) {
  EXPECT_EQ(greet("Alice"), "Hello Alice");
}

File: projects/java/greeter/src/test/java/com/github/racosta/greeter/GreeterTest.java
package com.github.racosta.greeter;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public final class GreeterTest {
  @Test
  public void testDefaultGreet() {
    Greeter greeter = new Greeter();
    org.junit.Assert.assertEquals("Hello World!", greeter.greet());
  }

  @Test
  public void testGreetWithName() {
    Greeter greeter = new Greeter();
    org.junit.Assert.assertEquals("Hello Bob!", greeter.greet("Bob"));
    org.junit.Assert.assertEquals("Hello Charlie!", greeter.greet("Charlie"));
  }
}
